pytd/client.py

```diff
@@ -57,7 +63,7 @@ def __init__(self, apikey=None, endpoint=None, database='sample_datasets', engin
 
         self.api_client = tdclient.Client(apikey=apikey, endpoint=endpoint, user_agent=engine.user_agent, **kwargs)
 
-        self.writer = None
+        self.writer = writer
```

Comment:
I need your opinion here.

This way, non-`None` `writer` instance created outside of `Client` will be closed at the time of `Client#close`. Do you think if it's an okay behavior? 

```py
writer = Writer()
client = Client(writer=writer)
client.close()
writer.write_dataframe(...)  # failed since it's already closed inside of client
```

Guessing...users may want to reuse `writer` instance even if `client` is closed. Same discussion should happen for the `engine` argument.

Reply:
ah, good point. 

There is a use case I can imagine that a writer could be reused; when a user creates a writer then the user uses it for both `Client` and `pandas_td`. In the case of that scenario, an engine also can be reused, but IMO the cost of creating `SparkWriter` is heavy rather than other Writes. It depends on who should manage `Writer`,  but this approach might introduce some complexity.

If you remind other use cases, I'd love to hear.

Options that I come up with are:
1) Keep it as is. A Writer will be closed and we can't reuse it.
2) Distinguish writer is created by Client or not, and close if the Writer was created by Client itself. This can be a little bit complex but we can reduce the cost of creation of SparkWriter.
3) Add some arguments something like `writer_options` which could pass extra option like `td_spark_path` (or `writer_options` can be more direct option; `td_spark_path`). This option should accept the cost of creation of SparkWriter.

I'm OK with option 3), which might be consistent with a future design change.

Reply:
I found I missed applying this change to `pandas_td`. After we dicide the design, I'll work on it.

Reply:
We will choose option 2, which is implemented in 69438a9 2dbc9f9